react/src/components/search-app/SearchBarComponent.jsx

```diff
@@ -0,0 +1,126 @@
+import Autocomplete from 'react-autocomplete';
+import React from 'react';
+
+
+export default class SearchBarComponent extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      query: props.query,
```

Comment:
You shouldn't put props into state. There should only be a single component that is responsible for the data and all child components should have read-only access via props. To manipulate the props from the child component, you would have to pass a callback as an additional property to the child component (this component in our case). See also [Lifting State Up](https://facebook.github.io/react/docs/lifting-state-up.html).